ts/src/main/scala/ch/epfl/ts/traders/BacktestTrader.scala

```diff
@@ -0,0 +1,105 @@
+package ch.epfl.ts.traders
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import akka.actor.Cancellable
+import ch.epfl.ts.component.{Component, ComponentRef}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Implements back testing for traders.
+  *
+  * To use this class, redirect all previous connections into trader to instances
+  * of this class. Connections out of the original trader should remain unchanged.
+  *
+  * @param trader the reference to the real trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the intial seed money
+  * @param period the period in milliseconds to compute performance
```

Comment:
In my Trader parameters I found it useful to take directly a `FiniteDuration` parameter rather than choosing a specific time unit in advance.